src/main/java/com/puppycrawl/tools/checkstyle/Checker.java

```diff
@@ -68,7 +68,8 @@
     public static final String EXCEPTION_MSG = "general.exception";
 
     /** Logger for Checker. */
-    private static final Log LOG = LogFactory.getLog(Checker.class);
+    @SuppressWarnings("PMD.LoggerIsNotStaticFinal")
+    private final Log log;
```

Comment:
@crud3 , please follow violations from CIs, all of them should be satisfied (green).

it is not allowed to use PMD suppressions by annotation.
We need to disable such PMD rule completely in our project as we are library and static is preferable in final applications only.

Reply:
Thanks for the feedback, I will get to work on it tomorrow!

Reply:
@romani I have removed the the PMD suppression annotations, squashed and rebased.
I also got rid of another Error by adding MainTest.java to the sevntu_suppressions.xml like Main.java
`    <!-- Main and MainTest are technically not part of the checkstyle library,
    so static Loggers are okay there. -->
`
`    <suppress checks="AvoidModifiersForTypesCheck"
              files="Main\.java|MainTest\.java"/>`
to suppress this rule in checkstyle_sevntu_checks.xml:
`<module name="AvoidModifiersForTypesCheck">`
`          <property name="forbiddenClassesRegexpProtected" value="Logger"/>`
`          <property name="forbiddenClassesRegexpPublic" value="Logger"/>`
`          <property name="forbiddenClassesRegexpPackagePrivate" value="Logger"/>`
`          <property name="forbiddenClassesRegexpStatic" value="Logger"/>`
`</module>`

Is that okay?

Next I could add the changes in checkstyle_sevntu_checks.xml you proposed, to see if they get rid of the remaining 8 errors.